### ISSUE TITLE
All windows ARM64 benchmark tests failing

### ISSUE THREAD

**Issue by LoopedBard3** [dotnet/performance#2421]
All (it seems) windows ARM64 benchmark tests are failing with the exception:
```
[2022/05/09 06:29:58][INFO] Process terminated. Infinite recursion during resource lookup within System.Private.CoreLib.  This may be a bug in System.Private.CoreLib, or potentially in certain extensibility points such as assembly resolve events or CultureInfo names.  Resource name: Arg_AccessViolationException
[2022/05/09 06:29:58][INFO]    at System.Environment.FailFast(System.String)
[2022/05/09 06:29:58][INFO]    at System.SR.InternalGetResourceString(System.String)
[2022/05/09 06:29:58][INFO]    at System.SR.GetResourceString(System.String)
[2022/05/09 06:29:58][INFO]    at System.AccessViolationException..ctor()
[2022/05/09 06:29:58][INFO]    at System.SpanHelpers.IndexOf(Char ByRef, Char, Int32)
[2022/05/09 06:29:58][INFO]    at System.String.Ctor(Char*)
[2022/05/09 06:29:58][INFO]    at System.Globalization.CultureData.GetLocaleInfoEx(System.String, UInt32)
[2022/05/09 06:29:58][INFO]    at System.Globalization.CultureInfo.GetUserDefaultLocaleName()
[2022/05/09 06:29:58][INFO]    at System.Globalization.CultureInfo..cctor()
[2022/05/09 06:29:58][INFO]    at System.Globalization.CultureInfo.get_CachedCulturesByName()
[2022/05/09 06:29:58][INFO]    at System.Globalization.CultureInfo.GetCultureInfo(System.String)
[2022/05/09 06:29:58][INFO]    at System.Resources.ManifestBasedResourceGroveler.GetNeutralResourcesLanguage(System.Reflection.Assembly, System.Resources.UltimateResourceFallbackLocation ByRef)
[2022/05/09 06:29:58][INFO]    at System.Resources.ResourceManager.CommonAssemblyInit()
[2022/05/09 06:29:58][INFO]    at System.SR.get_ResourceManager()
[2022/05/09 06:29:58][INFO]    at System.SR.InternalGetResourceString(System.String)
[2022/05/09 06:29:58][INFO]    at System.SR.GetResourceString(System.String)
[2022/05/09 06:29:58][INFO]    at System.AccessViolationException..ctor()
[2022/05/09 06:29:58][INFO]    at System.SpanHelpers.IndexOf(Char ByRef, Char, Int32)
[2022/05/09 06:29:58][INFO]    at System.String.Ctor(Char*)
[2022/05/09 06:29:58][INFO]    at System.AppContext.Setup(Char**, Char**, Int32)
```

**Comment by LoopedBard3** [comment]
cc @lewing We saw this same error found in https://github.com/dotnet/runtime/pull/67771, do you know if there is a different tracking issue or related investigation?

**Comment by lewing** [comment]
It looks like you are using your own perf machines for these builds, what version of windows is installed on them?

**Comment by LoopedBard3** [comment]
One of the two sets of failing Windows machines have version: Windows 10.0.19041, I am not sure about the other but think it would be similar.

**Comment by Claire-kangkang** [comment]
I also reproduce this issue with arm 64 testing, my machine is Windows 11 Enterprise Insider Preview (25096.1000). 

Reproduce steps: 
1. Install latest 7.0 sdk and hosting bundle. 
2. Delete everything in something like the 7.0.0-preview.xx.xx folder, and replace everything of the SDK installed version with the asp.net shared framework folder that was installed, i.e. 'C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\7.0.0-ci'
3. Run the project will ran into this issue.

**Comment by LoopedBard3** [comment]
This was fixed with the same fix as dotnet/runtime#69222 (issue tracking this on the runtime side).
